### ISSUE TITLE
[BUG] proto codec is not able to write errors

### ISSUE THREAD

**Issue by veger** [micro/go-micro#1902]
**Describe the bug**

1. What are you trying to do?

I am trying to return an error from my RPC service with a protobuf protocol (codec `proto`)

2. What did you expect to happen?

Error is encoded and written properly

3. What happens instead?

A hidden (see #1900) log message is written (i.e. nothing happens)

**How to reproduce the bug:**

Using v2.9.1, just return an error in from the protobuf handler
```golang

func MyHandler(ctx context.Context, req *proto.Request, rsp *proto.Response) error {
	return errors.New("this is an error")
}
```

I debug the code a little:
The hidden error:
https://github.com/micro/go-micro/blob/94bd1025a64eaaba1d89189891ef49f3ab5eee7f/server/rpc_server.go#L437

It is cause because
https://github.com/micro/go-micro/blob/94bd1025a64eaaba1d89189891ef49f3ab5eee7f/server/rpc_server.go#L425
fills in `nil` for `b` (second function parameter)

This ends up (via rpc_codec.go) in (so `b = nil`):
https://github.com/micro/go-micro/blob/94bd1025a64eaaba1d89189891ef49f3ab5eee7f/codec/proto/proto.go#L36

Which returns the (hidden) error.

When looking at JSON codec
https://github.com/micro/go-micro/blob/94bd1025a64eaaba1d89189891ef49f3ab5eee7f/codec/json/json.go#L34
it just returns (`nil`) in this case, preventing the error from occurring.

**Comment by veger** [comment]
Can we also have this fixed on the 2.9 branch?

**Comment by asim** [comment]
The 2.9 branch was just for the release

**Comment by veger** [comment]
Ok makes sense.
Can we instead have a v2.9.2 release with this fix and maybe some other interesting fixes (assuming there are)?
It is an issue for basically everyone using one of these codecs, and I suppose people will keep on using v2 for a while for their (existing) projects

**Comment by 0987363** [comment]
Any update?

The fix was v3, but the go-plugins still use v2.9.1.

**Comment by veger** [comment]
With the recent announcement and change of ownership/project intentions, I doubt it will be released in a v2.9.x release... :cry:
